File: Hydra/ObjectiveBee/taskE.py
import unittest

#### temp ####
class TestStringMethods(unittest.TestCase):
    def test1(self, res):
        self.assertEqual(res, "0 3 6")

    def test2(self, res):
        self.assertEqual(res, "0 0 4 11 11")
############

"""
ВХОД:
В первой строке дано целое число n (2≤n≤200000) — количество серверов.
Во второй строке через пробел записаны n целых чисел w_i (1 ≤ w_i ≤ 100000000) — вес i-го сервера.

ВЫХОД:
В единственной строке выведите через пробел n целых чисел res_i — суммарное количество энергии, необходимое для упорядочивания первых i серверов.
"""

"""
with open("input.txt") as file:
    pass

examples = ["3\n3 1 2", "5\n1 4 3 2 5"] # temp
"""
with open('input.txt', 'r') as fin:
    _ = fin.readline()
    ord = list(map(int, fin.readline().split(' ')))

ans = '0 '
mass = 0
for k in range(1, len(ord)):
    temp_ord = ord[0:k+1]
    operations = 0
    not_sorted = True
    while not_sorted:
        for j in range(len(temp_ord) - 1):
            if temp_ord[j] > temp_ord[j + 1]:
                temp_ord[j+1], temp_ord[j] = temp_ord[j], temp_ord[j+1]
                mass += temp_ord[j + 1]
                ord[j+1], ord[j] = ord[j], ord[j+1]
            elif j == len(temp_ord) - 2:
                not_sorted = False
            else:
                continue
    ans += str(mass) + ' '

ans = ans.strip()



testing = TestStringMethods()   # temp
# testing.test1(ans)          # temp
testing.test2(ans)    # temp

with open('output.txt', 'w') as fout:
    print(ans, file=fout)


"""
res = ""

with open("output.txt", "w") as file:
    file.write(res)
"""

File: Hydra/ObjectiveBee/taskE.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <sstream>
#include <vector>

using namespace std;

string n, str;
int len, mass = 0;

string ans = "0 ";

void read_input(void)
{
  ifstream fin("input.txt");

  getline(fin, n);
  getline(fin, str);

  len = stoi(n);
}

vector<int> split(string str)
{
  vector<int> result;
  istringstream iss(str);
  for (string str; iss >> str;)
    result.push_back(stoi(str));

  return result;
}

int main(void)
{
  int operations = 0;
  bool not_sorted = true;
  string temp;
  int temp_int;

  read_input();
  vector<int> ord = split(str);
  vector<int> temp_ord;

  temp_ord = vector<int>(ord.begin(), ord.begin() + 2);


  for (int i = 0; i < temp_ord.size(); i++)
    cout << temp_ord[i] << ' ';




  /*
  for (int i = 1; i < len; i++) {
    temp_ord =
    operations = 0;
    not_sorted = true;
    while (not_sorted) {
      for (int j = 0; j < (temp_ord.size() - 1); j++) {
        if (temp_ord[i] > temp_ord[j+1]) {
          temp = temp_ord[j+1];
          temp_ord[j+1] = temp_ord[j];
          temp_ord[j] = temp;
          mass += temp_ord[j+1];

          temp_int = ord[j+1];
          ord[j+1] = ord[j];
          ord[j] = temp_int;
        }
        else if (j == (temp_ord.size() - 2))
          not_sorted = false;
        else
          continue;
      }
      ans.append(to_string(mass) + " ");
    }
  }*/
  return 0;
}
